dspace-spring-rest/src/test/java/org/dspace/app/rest/MetadatafieldRestRepositoryIT.java

```diff
@@ -124,4 +163,190 @@ public void findByNullSchema() throws Exception {
+    }
+
+    @Test
+    public void deleteNonExisting() throws Exception {
+
+        MetadataField metadataField = createMetadataField();
+        deleteMetadataFieldIfExists();
+
+        Integer id = metadataField.getID();
+        assertThat(metadataFieldService.find(context, id), nullValue());
+
+        getClient(getAuthToken(admin.getEmail(), password))
+                .perform(delete("/api/core/metadatafields/" + id))
+                .andExpect(status().isNotFound());
+    }
+
+    @Test
+    public void update() throws Exception {
+
+        MetadataField metadataField = createMetadataField();
+
+        MetadataFieldRest metadataFieldRest = new MetadataFieldRest();
+        metadataFieldRest.setId(metadataField.getID());
+        metadataFieldRest.setElement(ELEMENT_UPDATED);
+        metadataFieldRest.setQualifier(QUALIFIER_UPDATED);
+        metadataFieldRest.setScopeNote(SCOPE_NOTE_UPDATED);
+
+        try {
+            getClient(getAuthToken(admin.getEmail(), password))
+                    .perform(put("/api/core/metadatafields/" + metadataField.getID())
+                            .content(new ObjectMapper().writeValueAsBytes(metadataFieldRest))
+                            .contentType(contentType))
+                    .andExpect(status().isOk());
+
+            metadataField = metadataFieldService.find(context, metadataField.getID());
+
+            assertEquals(ELEMENT_UPDATED, metadataField.getElement());
+            assertEquals(QUALIFIER_UPDATED, metadataField.getQualifier());
+            assertEquals(SCOPE_NOTE_UPDATED, metadataField.getScopeNote());
+        } finally {
+            deleteMetadataFieldIfExists(metadataField);
+        }
+    }
+
+    @Test
+    public void updateUnauthorized() throws Exception {
+
+        MetadataField metadataField = createMetadataField();
+
+        MetadataFieldRest metadataFieldRest = new MetadataFieldRest();
+        metadataFieldRest.setId(metadataField.getID());
+        metadataFieldRest.setElement(ELEMENT_UPDATED);
+        metadataFieldRest.setQualifier(QUALIFIER_UPDATED);
+        metadataFieldRest.setScopeNote(SCOPE_NOTE_UPDATED);
+
+        try {
+            getClient()
+                    .perform(put("/api/core/metadatafields/" + metadataField.getID())
+                            .content(new ObjectMapper().writeValueAsBytes(metadataFieldRest))
+                            .contentType(contentType))
+                    .andExpect(status().isUnauthorized());
+
+            metadataField = metadataFieldService.find(context, metadataField.getID());
+
+            assertEquals(ELEMENT, metadataField.getElement());
+            assertEquals(QUALIFIER, metadataField.getQualifier());
+            assertEquals(SCOPE_NOTE, metadataField.getScopeNote());
+        } finally {
+            deleteMetadataFieldIfExists(metadataField);
+        }
+    }
+
+    private MetadataField createMetadataField() throws AuthorizeException, SQLException, NonUniqueMetadataException {
```

Comment:
we should have a Builder for that

Reply:
This has been updated to use existing `MetadataFieldBuilder` class.